Assets/MRTK/Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -575,11 +575,60 @@ private void HandleFocusEvent(GameObject eventTarget, FocusEventData focusEventD
             // If event was not handled by modal, pass it on to the current focused object
             if (focusedObject != null)
             {
-                ExecuteEvents.ExecuteHierarchy(focusedObject, baseInputEventData, eventHandler);
+                // Handling propagation
+                var eventPropagationData = baseInputEventData as IMixedRealityEventPropagationData;
```

Comment:
Are there any events in MRTK that will not implement IMixedRealityEventPropagationData?

It will be important to document this distinction: "if an event does not implement IMixedRealityEventPropagationData, the event gets executed using Unity's Events.ExecuteHierarchy method". Otherwise I could see people getting very confused why some event dispatch behaves one way and some behaves another way.

Reply:
Yes. Since our issues were related to input events. Only Events that inherit from InputEventData are currently implementing the propagation interface. I see also need for focus events to include propagation data.